### DISCUSSION TITLE
Open temporal intervals

### DISCUSSION THREAD

**Discussion by rybesh** [periodo/edtf-ontology#13] [Ideas]
EDTF Level 1 provides expressions for *open* intervals of time, i.e. intervals that do not have a beginning, or that do not have an end. These are distinguished from intervals for which either the beginning or end is simply not known. The latter are easily modeled using the Time Ontology as time intervals for which the beginning or end is left unstated: under the open world assumption, if nothing is said about an endpoint of an interval, that just means we know nothing about it, not that it doesn’t exist.

Open intervals, however, are trickier, because of that same open world assumption. Simply omitting the missing endpoint would make open intervals indistinguishable from intervals with an unknown endpoint. Instead, we want to express that the interval *does not have* an endpoint, not that we do not know anything about the endpoint.

For this purpose the draft EDTF Ontology defines two subclasses of [`time:ProperInterval`](https://www.w3.org/TR/owl-time/#time:ProperInterval) for modeling open intervals:

* [`edtfo:OpenBeginningInterval`](https://periodo.github.io/edtf-ontology/#Openbeginninginterval)
* [`edtfo:OpenEndInterval`](https://periodo.github.io/edtf-ontology/#Openendinterval)

The former adds the constraint that instances must have zero [`time:hasBeginning`](https://www.w3.org/TR/owl-time/#time:hasBeginning) values, while the latter adds the constraint that instances must have zero [`time:hasEnd`](https://www.w3.org/TR/owl-time/#time:hasEnd) values.

Examples:

* [level-1/open-end-interval-day-precision.ttl](https://github.com/periodo/edtf-ontology/blob/main/cases/level-1/open-end-interval-day-precision.ttl)
* [level-1/open-start-interval-day-precision.ttl](https://github.com/periodo/edtf-ontology/blob/main/cases/level-1/open-start-interval-day-precision.ttl)

**Comment by nichtich** [comment]
Looks like this has been implemented and examples at

- https://github.com/periodo/edtf-ontology/tree/main/cases/level-1/interval/open-start
- https://github.com/periodo/edtf-ontology/tree/main/cases/level-1/interval/open-end

So this conversation can be closed as resolved.
